metis-normalization/src/test/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractorTest.java

```diff
@@ -0,0 +1,68 @@
+package eu.europeana.normalization.dates.extraction.dateextractors;
+
+import static eu.europeana.normalization.dates.DateNormalizationExtractorMatchId.DECADE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.params.provider.Arguments.of;
+
+import eu.europeana.normalization.dates.DateNormalizationExtractorMatchId;
+import eu.europeana.normalization.dates.DateNormalizationResult;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class DecadeDateExtractorTest {
+
+  private static final DecadeDateExtractor DECADE_DATE_EXTRACTOR = new DecadeDateExtractor();
+
+  @ParameterizedTest
+  @MethodSource
+  void extract(String input, String expected, DateNormalizationExtractorMatchId dateNormalizationExtractorMatchId) {
+    final DateNormalizationResult dateNormalizationResult = DECADE_DATE_EXTRACTOR.extract(input);
+    if (expected == null) {
+      assertNull(dateNormalizationResult);
+    } else {
+      final String actual = dateNormalizationResult.getEdtfDate().toString();
+      assertEquals(expected, actual);
+      if (actual.contains("?")) {
+        assertTrue(dateNormalizationResult.getEdtfDate().isUncertain());
+      } else {
+        assertFalse(dateNormalizationResult.getEdtfDate().isUncertain());
+      }
+      assertEquals(dateNormalizationExtractorMatchId, dateNormalizationResult.getDateNormalizationExtractorMatchId());
+    }
+  }
+
+  private static Stream<Arguments> extract() {
+    return Stream.of(
+        of("180x", "180X", DECADE),
+        of("180u", "180X", DECADE),
+        of("180X", "180X", DECADE),
+        of("180U", "180X", DECADE),
+        of("  180u  ", "180X", DECADE),
+        of("180x?", "180X?", DECADE),
+        of("180u?", "180X?", DECADE),
+        of("180??", "180X?", DECADE),
+        of("?180x", "180X?", DECADE),
+        of("?180u", "180X?", DECADE),
+        of("?180u?", "180X?", DECADE),
```

Comment:
Derived from the regex new test cases came up.
```
        of("?180x?", "180X?", DECADE),
        of("?180??", "180X?", DECADE),

        //just to validate that for 3 digits the decade is extracted at this moment.
        of("?1280x", null, null),
        of("?1280u?", null, null),
        of("?1280??", null, null),
        of("1280??", null, null),
```

Reply:
I've added the cases you'd like